api/graphql/tests/test_reservations/test_reservation_create.py

```diff
@@ -406,6 +406,32 @@ def test_create_fails_when_reservation_unit_closed_on_selected_time(
             content.get("data").get("createReservation").get("errors")[0]["messages"][0]
         ).contains("Reservation unit is not open within desired reservation time.")
 
+    def test_create_succeed_when_reservation_unit_closed_on_selected_time_and_opening_hours_are_ignored(
+        self, mock_periods, mock_opening_hours
+    ):
+
+        self.reservation_unit.allow_reservations_without_opening_hours = True
+        self.reservation_unit.save()
+
+        mock_opening_hours.return_value = self.get_mocked_opening_hours()
+        input_data = self.get_valid_input_data()
+        today = datetime.date.today()
+        begin = datetime.datetime(today.year, today.month, today.day, 21, 0)
+        end = begin + datetime.timedelta(hours=2)
+        input_data["begin"] = begin.strftime("%Y%m%dT%H%M%SZ")
+        input_data["end"] = end.strftime("%Y%m%dT%H%M%SZ")
+
+        self.client.force_login(self.regular_joe)
+        response = self.query(self.get_create_query(), input_data=input_data)
+        content = json.loads(response.content)
+
+        assert_that(content.get("errors")).is_none()
+        assert_that(
+            content.get("data").get("createReservation").get("errors")
+        ).does_not_contain(
+            "Reservation unit is not open within desired reservation time."
+        )
+
```

Comment:
We should add here a check that the actual reservation has been created.

Reply:
Added! 👍